akka-actor/src/main/scala/akka/io/Inet.scala

```diff
@@ -16,38 +19,51 @@ object Inet {
     /**
      * Action to be taken for this option before bind() is called
      */
-    def beforeBind(ds: DatagramChannel): Unit = ()
+    def beforeDatagramBind(ds: DatagramSocket): Unit = ()
 
     /**
      * Action to be taken for this option before bind() is called
      */
-    def beforeBind(ss: ServerSocketChannel): Unit = ()
+    def beforeServerSocketBind(ss: ServerSocket): Unit = ()
 
     /**
-     * Action to be taken for this option before bind() is called
+     * Action to be taken for this option before calling connect()
      */
-    def beforeBind(s: SocketChannel): Unit = ()
+    def beforeConnect(s: Socket): Unit = ()
 
     /**
      * Action to be taken for this option after connect returned (i.e. on
      * the slave socket for servers).
      */
-    def afterConnect(c: DatagramChannel): Unit = ()
+    def afterConnect(s: Socket): Unit = ()
+  }
+
+  /**
+   * Java API: AbstractSocketOption is a package of data (from the user) and associated
+   * behavior (how to apply that to a channel).
+   */
+  abstract class AbstractSocketOption extends SocketOption
 
+  trait AfterChannelConnect extends SocketOption {
     /**
      * Action to be taken for this option after connect returned (i.e. on
      * the slave socket for servers).
      */
-    def afterConnect(c: ServerSocketChannel): Unit = ()
+    def afterConnect(c: DatagramChannel): Unit = ()
```

Comment:
This is confusing (well, it was already before your change): this applies only “after bind” but is named “after connect”, while the differentiator in the method signature is the socket/channel dichotomy. Since this did not exist in 2.3 we should rename this to `afterBind` instead.